### PR TITLE
Spurious deactivation of channels

### PR DESCRIPTION
**Pull request by franckHERESON**
Correction of  registers addresses and bit manipulation that deactivate channels on AD7771.

### CODE REVIEW

**Comment by mirceacaprioru** [comment]
The modifications look good just a few remarks.

Could you split the commit into two for each modification and also add Signed-off-by tag. Also remove the merge commit from the pull request.

File: drivers/adc/ad7779/ad7779.c [hunk 1]
```diff
@@ -425,8 +425,8 @@ int32_t ad7779_set_state(ad7779_dev *dev,
 
 	ret = ad7779_spi_int_reg_write_mask(dev,
 					    AD7779_REG_CH_DISABLE,
-					    AD7779_CH_DISABLE(0x1),
-					    AD7779_CH_DISABLE(state));
+					    AD7779_CH_DISABLE(ch),
+					    state ? AD7779_CH_DISABLE(ch) : 0x00);
```
Comment (dbogdan) [inline drivers/adc/ad7779/ad7779.c]: I wouldn't use the hexadecimal representation - for consistency with the rest of code:
state ? AD7779_CH_DISABLE(ch) : 0);

**Comment by franckHERESON** [comment]
Fine for me.

 

Best Regards/Cordialement,

F.HÉRÉSON 

<mailto:franck.hereson@secad.fr> 

http://www.secad.fr <http://www.secad.fr/>  

SECAD Grenoble 34 Avenue de l'Europe 38100 GRENOBLE 

Tel + 33 (0) 4 76 33 05 21 - Fax +33 (0) 4 76 33 05 56 

 

 

 

De : Dragos Bogdan [mailto:notifications@github.com] 
Envoyé : vendredi 27 septembre 2019 16:37
À : analogdevicesinc/no-OS <no-OS@noreply.github.com>
Cc : franckHERESON <franck.hereson@secad.fr>; Author <author@noreply.github.com>
Objet : Re: [analogdevicesinc/no-OS] Spurious deactivation of channels (#250)

 

@dbogdan requested changes on this pull request.

  _____  

In drivers/adc/ad7779/ad7779.c <https://github.com/analogdevicesinc/no-OS/pull/250#discussion_r329103122> :

> @@ -425,8 +425,8 @@ int32_t ad7779_set_state(ad7779_dev *dev,
 
        ret = ad7779_spi_int_reg_write_mask(dev,
                                          AD7779_REG_CH_DISABLE,
-                                         AD7779_CH_DISABLE(0x1),
-                                         AD7779_CH_DISABLE(state));
+                                         AD7779_CH_DISABLE(ch),
+                                         state ? AD7779_CH_DISABLE(ch) : 0x00);

I wouldn't use the hexadecimal representation - for consistency with the rest of code:
state ? AD7779_CH_DISABLE(ch) : 0);

**Comment by dbogdan** [comment]
Can you please update the pull request according to the suggestions so it can be merged?
Thanks,
Dragos

**Comment by dbogdan** [comment]
Since there was no recent activity, this pull request will be closed. https://github.com/analogdevicesinc/no-OS/pull/407 will fix the issue.

**Pull request closed without merging**